crates/proof_of_stake/src/epoched.rs

```diff
@@ -106,7 +107,7 @@ where
             Some(last_update) => {
                 let data_handler = self.get_data_handler();
                 let future_most_epoch =
-                    last_update + FutureEpochs::value(params);
+                    last_update.unchecked_add(FutureEpochs::value(params));
```

Comment:
why unchecked add?

Reply:
This is for convenience when working with epochs - it's u64 so we should never overflow on additions with system params. I added this doc on the method: Panics on overflow. Care must be taken to only use this with trusted values that are known to be in a limited range (e.g. system parameters but not e.g. transaction variables).